janus_server/src/bin/janus_cli.rs

```diff
@@ -56,17 +56,15 @@ enum Command {
     },
 
     /// Create a datastore key and write it to a Kubernetes secret.
-    CreateDatastoreKey {
-        /// The Kubernetes namespace to create the datastore key secret in.
-        k8s_namespace: String,
-
-        /// The name of the Kubernetes secret to place the datastore key in.
-        k8s_secret_name: String,
-    },
+    CreateDatastoreKey,
 }
 
 impl Command {
     async fn execute(&self, options: &Options, config: &Config) -> Result<()> {
+        let k8s_client = kube::Client::try_default()
+            .await
+            .context("couldn't connect to Kubernetes environment")?;
```

Comment:
With `janus_cli decode-dap-message` on the way, we should avoid constructing this client unless it is required by the subcommand, so that `write-schema` and `decode-dap-message` can be used on systems without kubeconfig files. `once_cell::unsync::Lazy` would be a convenient way to do so without duplication.

Reply:
+1, we should maintain that we don't access/connect to dependencies that are not used for the particular subcommand.